### DISCUSSION TITLE
heads up: breaking change  quasar 2.7.7.+ i18n + CompositionAPI

### DISCUSSION THREAD

**Discussion by XVonRad** [quasarframework/quasar#14135] [General - Components / Directives / etc]
Since the latest package upgrade (quasar 2.7.7), vue-i18n throws an exception on encountering
`useI18n({ useScope: 'global' });` when used with CompositionAPI
To fix it, add `allowComposition: true` in `src/boot/i18n.js`. The docs do not mention this detail ...yet

**Comment by yusufkandemir** [comment]
The `quasar` package isn't remotely related to `vue-i18n`, you can check the code to confirm. `@quasar/app-vite` and `@quasar/app-webpack` just set some build time optimization flags and configs. So, this is not a breaking change on Quasar's side, in fact, there are no updates about i18n. You probably updated other packages as well (_intentionally or unintentionally_). So, the tip for vue-i18n users is appreciated, but I wanted to state that this is not related to Quasar.

Previously, you had to set `legacy: false` to use Composition API, but now that option makes it possible to use `useI18n` even with `legacy: true`. Probably a bug in vue-i18n previously allowed you to use `useI18n` even if you had `legacy: false`, but now they've fixed it and put it behind `allowComposition: true`. If that's not the case, then it's another type of bug. So, please refer to vue-i18n change logs and community for these types of things next time, thanks.

**Reply by yusufkandemir** [reply]
However, if you think if any problem including this is caused by Quasar, please make a repro so we can take a look.

**Reply by XVonRad** [reply]
You're right, nothing to do with quasar as such. I just want to point out that it would be useful to add this updated requirement to the quasar docs @ https://quasar.dev/options/app-internationalization, as the problem will affect quasar users who, like me, base their config on  recommendations found in said docs.
If this is not the right place for reporting that sort of thing, please tell me where...

**Reply by yusufkandemir** [reply]
Please create a docs report for this. You can do so by going to the issues tab, clicking the "new issue" button, then choosing the related option.
![image](https://user-images.githubusercontent.com/6266078/182821865-79d23f62-a547-4984-bc84-fd0f2d218e52.png)

Please also share the full code, what you've tried before, the version of vue-i18n the solution is targeted for, etc. so we can update the docs correctly and in detail. Thanks 👏
